### PR TITLE
Image: add support for webp

### PR DESCRIPTION
**Pull request by enumag**
Not sure how to write tests - there aren't any tests specific for the other types either.

Also should I check for php version?

### CODE REVIEW

File: src/Utils/Image.php [hunk 1]
```diff
@@ -101,7 +101,8 @@ class Image extends Nette\Object
 	/** image types */
 	const JPEG = IMAGETYPE_JPEG,
 		PNG = IMAGETYPE_PNG,
-		GIF = IMAGETYPE_GIF;
+		GIF = IMAGETYPE_GIF,
+		WEBP = -1;
```
Comment (enumag) [inline src/Utils/Image.php]: Similar constant for WEBP does not exist (yet).

File: src/Utils/Image.php [hunk 2]
```diff
@@ -576,10 +583,10 @@ public function __toString()
 	 */
 	public function send($type = self::JPEG, $quality = NULL)
 	{
-		if (!in_array($type, [self::JPEG, self::PNG, self::GIF], TRUE)) {
+		if (!in_array($type, [self::JPEG, self::PNG, self::GIF, self::WEBP], TRUE)) {
 			throw new Nette\InvalidArgumentException("Unsupported image type '$type'.");
 		}
-		header('Content-Type: ' . image_type_to_mime_type($type));
+		header('Content-Type: ' . ($type === self::WEBP ? 'image/webp' : image_type_to_mime_type($type)));
```
Comment (enumag) [inline src/Utils/Image.php]: Webp is not supported by image_type_to_mime_type.

**Comment by dg** [comment]
Does getimagesizefromstring support WEBP?

**Comment by enumag** [comment]
Oh boy of course it does not. :-(

**Comment by enumag** [comment]
There is no way to detect webp images in fromString and fromFile (aside from checking file extension in fromFile) so I removed webp from both. But I still think Nette\Utils\Image should be able to generate webp images.

**Comment by dg** [comment]
It is possible via `$image->webp('file.webp');` Full support is currently IMHO not possible.

**Comment by enumag** [comment]
Ok, closing for now. Thank you.

**Comment by enumag** [comment]
Webp support has been added for PHP 7.1RC4. It should be possible to implement this now.

**Pull request closed without merging**